narwhals/_spark_like/expr_dt.py

```diff
@@ -62,3 +65,14 @@ def _weekday(_input: Column) -> Column:
             return (self._compliant_expr._F.dayofweek(_input) + 6) % 7
 
         return self._compliant_expr._with_callable(_weekday)
+
+    def truncate(self, every: str) -> SparkLikeExpr:
+        multiple, unit = parse_interval_string(every)
+        if multiple != 1:
+            msg = f"Only multiple 1 is currently supported for Spark-like.\nGot {multiple!s}."
+            raise ValueError(msg)
+
+        def _truncate(_input: Column) -> Column:
+            return self._compliant_expr._F.date_trunc(UNITS_DICT[unit], _input)
```

Comment:
Very minor, but we can reuse the lookup across multiple calls this way

```suggestion
        format = UNITS_DICT[unit]

        def _truncate(_input: Column) -> Column:
            return self._compliant_expr._F.date_trunc(format, _input)
```